packages/components/src/styles/components/app-header.scss

```diff
@@ -0,0 +1,119 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+//
+// APP-HEADER
+//
+
+@use "../mixins/interactive-dark-theme" as *;
+
+// ----------------
+// Q: Create tokens?
+$app-header-home-link-padding: 4px;
+
+// Breakpoints WIP: START ----------------------------
+
+// Break points
+$tablet-min-width: 768px;
+$desktop-min-width: 1024px;
+
+// base styles
+.hds-app-header__menu {
+  display: inherit; // may be over-riddden by queries
+  gap: inherit;
+}
+
+// Media queries
+// -------------
+.hds-app-header {
+  // Mobile/small and tablet/medium compound view: applies at tablet min-width and below
+  @media (max-width: #{$desktop-min-width - 1}) {
+    &:not(.hds-app-header__menu-is-open) {
+      .hds-app-header__menu {
+        // dropdown menu is hidden when isOpen = false
+        display: none;
+      }
+    }
+
+    // Display content as dropdown menu in mobile/tablet view
+    .hds-app-header__menu {
+      position: absolute;
+      top: calc(100% + 1px); // We add 1px to account for the bottom border
+      right: 0;
+      left: 0;
+      z-index: 5; // TODO: determine correct z-index to use
+      display: flex;
+      flex-wrap: wrap;
+      // height: calc(100vh - 52px); // it should take the full height of the viewport minus the header height
+      padding: 16px;
+      background: var(--token-color-palette-neutral-700);
+    }
+
+    // Force nested dropdown content to align left in mobile/tablet view
+    .hds-dropdown__content {
+      left: 0;
+    }
+
+    // Hide :globalAfter content in mobile/tablet view
+    .hds-app-header__global-items-mobile-view {
+      display: none;
+    }
+
+    // Since the same component is used for both mobile/tablet and desktop views, we need to force the width to 100% in mobile view
+    .hds-app-header__global-items-desktop-view {
+      &,
+      .hds-dropdown,
+      .hds-dropdown-toggle-button {
+        width: 100%;
+      }
+    }
+  }
+
+  // Desktop/large view: applies at desktop min-width and above
+  @media (min-width: #{$desktop-min-width}) {
+    // Hide Menu button & Global Items after content in desktop view
+    .hds-app-header__menu-button,
+    .hds-app-header__global-items-desktop-view {
+      display: none;
+    }
+  }
+}
+
+// Breakpoints WIP: END ------------------------------
+
+.hds-app-header {
+  position: relative;
+  display: flex;
+  gap: 12px;
```

Comment:
Just so I'm following here, setting this at the root level of the component cascades down to the child elements to enforce a consistent gap in _all_ nested child elements?

Reply:
It does wherever I'm telling it to inherit the style (`gap: inherit;`)